include/sos/barriers/WeightedTwoSidedBarrier.h

```diff
@@ -0,0 +1,148 @@
+// VolEsti (volume computation and sampling library)
+
+// Copyright (c) 2012-2020 Vissarion Fisikopoulos
+// Copyright (c) 2018-2020 Apostolos Chalkis
+// Copyright (c) 2022-2022 Ioannis Iakovidis
+
+// Contributed and/or modified by Ioannis Iakovidis, as part of Google Summer of
+// Code 2022 program.
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+// References
+// Yunbum Kook, Yin Tat Lee, Ruoqi Shen, Santosh S. Vempala. "Sampling with
+// Riemannian Hamiltonian
+// Monte Carlo in a Constrained Space"
+
+// The log barrier for the domain {lu <= x <= ub}:
+//	phi(x) = - sum log(x - lb) - sum log(ub - x).
+#ifndef WEIGHTEDTWOSIDEDBARIER_H
+#define WEIGHTEDTWOSIDEDBARIER_H
+
+#include "Eigen/Eigen"
+#include "cartesian_geom/cartesian_kernel.h"
+#include <vector>
+
+template <class Point> class WeightedTwoSidedBarrier {
+
+  using NT = typename Point::FT;
+  using MT = Eigen::Matrix<NT, Eigen::Dynamic, Eigen::Dynamic>;
+  using VT = Eigen::Matrix<NT, Eigen::Dynamic, 1>;
+
+public:
+  VT lb;
+  VT ub;
+  int vdim;
+  int n;
+  std::vector<int> upperIdx;
+  std::vector<int> lowerIdx;
+  std::vector<int> freeIdx;
+  VT center;
+  const NT max_step = 1e16; // largest step size
+  VT extraHessian;
+
+  const NT inf = std::numeric_limits<NT>::infinity();
+
+  VT w;
+
+  WeightedTwoSidedBarrier(VT const &_lb, VT const &_ub, VT const &_w,
+                          int _vdim = 1) {
+    set_bound(_lb, _ub);
+    w = _w;
+    vdim = _vdim;
+    extraHessian = (1e-20) * VT::Ones(n);
+  }
+  WeightedTwoSidedBarrier() { vdim = 1; }
+
+  VT gradient(VT const &x) {
+    return w.cwiseQuotient(ub - x) - w.cwiseQuotient(x - lb);
+  }
+
+  VT hessian(VT const &x) {
+    VT d = w.cwiseQuotient((ub - x).cwiseProduct((ub - x))) +
+           w.cwiseQuotient((x - lb).cwiseProduct((x - lb)));
+    return d + extraHessian;
+  }
+  VT tensor(VT const &x) {
+    VT d = 2 * w.cwiseQuotient(
```

Comment:
I think it would be more clear in one line i.e. 
```
VT d = 2 * w.cwiseQuotient(((ub - x).cwiseProduct((ub - x))).cwiseProduct((ub - x))) -
```

Reply:
ok changed in both files